Add CLI parameters for device name and hex output path

solidspark opened the pull request:
## Changes

- Add `--name`/`-n` parameter to specify device name (auto-generates if not provided)
- Add `--file`/`-f` parameter to specify hex file output path for generate-hex method
- Update `convert_elf_to_hex` to use `output_path` parameter with proper path handling
- Improve hex file path display to show absolute path
- Bump version to 0.0.6

## Usage Examples

```bash
# With custom name and custom output file
hubbledemo flash lp_em_cc2340r5 -n my-device -f /path/to/firmware.hex -o ORG_ID -t TOKEN

# With custom name and default output location
hubbledemo flash lp_em_cc2340r5 -n my-device -o ORG_ID -t TOKEN

# With auto-generated name
hubbledemo flash lp_em_cc2340r5 -o ORG_ID -t TOKEN
```

## Testing

Tested with TI development boards (lp_em_cc2340r5) to ensure proper hex file generation with custom names and output paths.

solidspark committed:
Add CLI parameters for device name and hex output path

- Add --name/-n parameter to specify device name (auto-generates if not provided)
- Add --file/-f parameter to specify hex file output path
- Update convert_elf_to_hex to use output_path parameter with proper path handling
- Improve hex file path display to show absolute path
- Bump version to 0.0.6

buckleypaul commented on python/src/hubbledemo/cli.py:
```diff
@@ -119,10 +135,24 @@ def flash(board: str, name: str = None, org_id: str = None, token: str = None) -
+        
+        # Determine output path
+        if file:
+            output_path = file
```
We need to enforce a .hex extension else you can't open it in UniFlash - so we should check if it's not there and add if not.

Edit: I see that's in the convert_elf_to_hex function - IMO it should just be here so that we also echo the correct file name in line 155 (else it won't show the file extension there).

buckleypaul commented on python/src/hubbledemo/cli.py:
```diff
@@ -119,10 +135,24 @@ def flash(board: str, name: str = None, org_id: str = None, token: str = None) -
+        if file:
+            output_path = file
+        else:
+            output_path = f"{name}.hex"
```
```name``` will get overwritten by line 115 - that line needs to be updated as well with the introduction of the function param.

solidspark committed:
Fix name parameter overwriting issue

- Use separate device_name variable instead of overwriting name parameter
- Preserve original name parameter for hex filename logic
- Ensures user intent is respected for both device naming and file output
